jar-service/src/main/java/com/sixsq/slipstream/user/UserResource.java

```diff
@@ -91,16 +95,30 @@ protected boolean isMachineAllowedToAccessThisResource(){
 		return true;
 	}
 
-	private void mergeCloudSystemParameters(User user)
+	private void replaceUserPublicSshKeyWithServerSshPublicKeyIfEmpty(User user)
 			throws ConfigurationException, ValidationException {
+
+		Cookie cookie = CookieUtils.extractAuthnCookie(getRequest());
+
+		if (CookieUtils.isMachine(cookie) == true) {
+			UserParameter userPublicSshKey = user.getParameter(ExecutionControlUserParametersFactory.CATEGORY + "."
+					+ UserParametersFactoryBase.SSHKEY_PARAMETER_NAME);
+
+			if (userPublicSshKey != null && userPublicSshKey.getValue().trim().isEmpty()) {
```

Comment:
I don't think so because if I do that, we will enter into the `if` if `userPublicSshKey` is null and then we will get a NullPointerException because of `userPublicSshKey.setValue(serverPublicSshKey)` and `userPublicSshKey.getName()`.
@loomis Do you agree ?